lua/nvim-tree/utils.lua

```diff
@@ -11,6 +12,14 @@ function M.echo_warning(msg)
   api.nvim_command('echohl None')
 end
 
+function M.read_file(path)
+  local fd = uv.fs_open(path, "r", 438)
+  local stat = uv.fs_fstat(fd)
+  local data = uv.fs_read(fd, stat.size, 0)
```

Comment:
these functions fails if a file is not found. Doing checks on each line would be necessary imo:
```lua
if not fd then return '' end

if not stat then return '' end

return data or ''
```